app/src/main/java/com/example/readbitso/BitsoViewmodel.kt

```diff
@@ -0,0 +1,305 @@
+                price=it.price.take(10)
+            ))
+        }
+        isloading = returnlist.isNotEmpty()
+        return returnlist
+    }
+
+    //------------
+    //DataStore
+    fun setCoin(coin: String) {
+        viewModelScope.launch {
+            bitsoRepositoryImp.selectCoin("name",coin)
+        }
+    }
+
+    private suspend inline fun getCoin(): String? {
+        viewModelScope.launch {
+            selectedCoin=bitsoRepositoryImp.getCoin("name")
+        }
+        if(errormessage=="")
+            lastconsume=tokens(selectedCoin.toString())
+
+
+        return selectedCoin
+    }
+
+
+    fun selectPage(Page: String) = viewModelScope.launch {
+        bitsoRepositoryImp.setPage("page", Page)
+    }
+
+    private suspend inline fun getPage(): String? {
+        viewModelScope.launch{
+            page= bitsoRepositoryImp.getPage("page")
+        }
+        return page
+    }
+
+
+    //room
+    //----------------------
+    private suspend inline fun insertdbTokens() =
+        bitsoRepositoryImp.insertBooks(openedPayloads)
+
+
+
+    private suspend fun getbdTokens(): MutableList<BooksPayload> {
+        val returnlist = mutableListOf<BooksPayload>()
+
+        bitsoRepositoryImp.getflowBooks()
+            .collect {
+                it.forEach { moneda->
+                    returnlist.add(BooksPayload(
+                        book = moneda.book,
+                        id = moneda.id,
+                        maximum_price = moneda.maximum_price,
+                        minimum_price=moneda.minimum_price))
+                }
+            }
+        return returnlist
+    }
+
+
+    private suspend inline fun insertdbTrades() =
+        bitsoRepositoryImp.insertTrades(trades)
+
+
+    private suspend inline fun getdbTrades(): MutableList<PayloadTrades> {
+        val returnlist = mutableListOf<PayloadTrades>()
+
+        bitsoRepositoryImp.getflowTrades()
+            .catch {  }
+            .collect { it ->
+                it.forEach {
+                    returnlist.add(PayloadTrades(
+                        amount=it.Amount.toString(),
+                        book = it.pair.toString(),
+                        maker_side = operationKind(it.Type.toString()),
+                        price = it.Price.toString(),
+                    ))
+                }
+                trades=returnlist
+            }
+        return returnlist
+    }
+
+    private suspend inline fun insertdbAskBids() =
+        bitsoRepositoryImp.insertAsk(openedPayloadsCoin)
+
+
+    private suspend inline fun getdbAskBids()
+    {
+        val returnlist = mutableListOf<PayloadTickers>()
+        bitsoRepositoryImp.getflowAskBids()
+            .catch {  }
+            .collect { it ->
+                it.forEach {
+                    returnlist.add(PayloadTickers(
+                        ask = it.Ask,
+                        bid=it.Bid,
+                        book=it.Book
+                    ))
+                }
+            }
```

Comment:
Puedes hacer uso del operador `map`
```suggestion
        bitsoRepositoryImp.getflowAskBids()
            .catch {  }
            .map{
            PayloadTickers(
                        ask = it.Ask,
                        bid=it.Bid,
                        book=it.Book
                    )
            }
            .collect { it ->
                returnlist.addAll(it)
            }
```